### PR TITLE
Manu WIP

### PR DESCRIPTION
**Pull request by manukj**
completed Assignment 1,2,3,4

### CODE REVIEW

File: Q-1,2,3,4/src/Assignment1/Items.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		} else if (type.equals("manufactured")) {
+			// manufactured: 12.5% of the item cost + 2% of (item cost + 12.5% of the item
+			// cost)
+			this.tax = (double) ((0.125) * price) + (0.2 * (price + (0.125 * price)));
+		} else if (type.equals("imported")) {
+			import_tax = (double) (0.1) * price;
+			tot_price = price + import_tax;
+			double super_charge = (tot_price <= 100) ? 5 : (tot_price <= 200) ? 10 : 0.5 * (tot_price);
+			this.tax = import_tax + super_charge;
+		}
```
Comment (abhinav-adtechs) [inline Q-1,2,3,4/src/Assignment1/Items.java:47]: Consider leveraging java's inheritance property to make the code cleaner and scalable.

File: Q-1,2,3,4/src/Assignment1/TestAssignment1.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+
+class TestAssignment1 {
+
+	@Test
+	void test_for_type_row() {
```
Comment (abhinav-adtechs) [inline Q-1,2,3,4/src/Assignment1/TestAssignment1.java:16]: Must write test for edge cases also.

File: Q-1,2,3,4/src/Assignment2/StudentController.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	}
+
+	public void deleteStudent(int rollNumber) {
+		for (Student student : listOfStudents) {
+			if (student.getRollNumber() == rollNumber) {
+				listOfStudents.remove(student);
+				studentsRollNumber.remove(student.getRollNumber());
+				return;
+			}
+		}
```
Comment (abhinav-adtechs) [inline Q-1,2,3,4/src/Assignment2/StudentController.java]: Consider using streams API

File: Q-1,2,3,4/src/Assignment2/StudentController.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	// function to display the user
+	public void disaplayAlllistOfStudents(int c) {
+		if (listOfStudents.size() != 0) {
+			Student[] students = new Student[listOfStudents.size()];
+			students = listOfStudents.toArray(students);
+			if (c == 1) {
+				Arrays.sort(students);
+			} else {
+				Arrays.sort(students, Collections.reverseOrder());
+			}
```
Comment (abhinav-adtechs) [inline Q-1,2,3,4/src/Assignment2/StudentController.java]: Can we avoid this check?

File: Q-1,2,3,4/src/Assignment2/TestAssignment2.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import org.junit.jupiter.api.Test;
+
+@DisplayName("Assignment 2")
+class TestAssignment2 {
+
+	@Test
+	@Order(1)
+	@DisplayName("Validating String and Empty field")
+	void test_String_and_Empty_Field() {
+		Input input = new Input();
```
Comment (abhinav-adtechs) [inline Q-1,2,3,4/src/Assignment2/TestAssignment2.java:21]: Expect more test cases.

File: Q-1,2,3,4/src/Assignment3/DependencyGraph.java [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+ * Created by Manu KJ 
+ */
+package Assignment3;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+public class DependencyGraph {
```
Comment (abhinav-adtechs) [inline Q-1,2,3,4/src/Assignment3/DependencyGraph.java:11]: We can improve on code structure.

File: Q-1,2,3,4/src/Assignment4/DataBase.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+	public ResultSet getResultSet() {
+		return resultSet;
+	}
+
+	public int getTotalRow() {
+		return totalRow;
+	}
+
+	public boolean connectToDatabase() {
```
Comment (abhinav-adtechs) [inline Q-1,2,3,4/src/Assignment4/DataBase.java:40]: Consider making it singleton.

File: Q-1,2,3,4/src/Assignment4/Items.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+		} else if (type.equals("manufactured")) {
+			// manufactured: 12.5% of the item cost + 2% of (item cost + 12.5% of the item
+			// cost)
+			this.tax = (double) ((0.125) * price) + (0.2 * (price + (0.125 * price)));
+		} else if (type.equals("imported")) {
+			import_tax = (double) (0.1) * price;
+			tot_price = price + import_tax;
+			double super_charge = (tot_price <= 100) ? 5 : (tot_price <= 200) ? 10 : 0.5 * (tot_price);
+			this.tax = import_tax + super_charge;
+		}
```
Comment (abhinav-adtechs) [inline Q-1,2,3,4/src/Assignment4/Items.java:45]: Consider using java inheritance property

**Review by abhinav-adtechs — COMMENTED** [review]
Ok. 
Need not add binary files in git

File: Q-1,2,3,4/src/Assignment2/StudentController.java (hunk 4 above)
Comment (manukj) [reply Q-1,2,3,4/src/Assignment2/StudentController.java]: oh yeah, I misunderstood the question I thought the display is either in ascending or descending order

File: Q-1,2,3,4/src/Assignment2/StudentController.java (hunk 3 above)
Comment (manukj) [reply Q-1,2,3,4/src/Assignment2/StudentController.java]: yeah done ....

File: Q-1,2,3,4/src/Assignment4/DataBase.java (hunk 7 above)
Comment (manukj) [reply Q-1,2,3,4/src/Assignment4/DataBase.java:40]: done
